core/src/common/cdk-deployer-build.ts

```diff
@@ -1,15 +1,101 @@
+export const startBuild = `
+const respond = async function(event, context, responseStatus, responseData, physicalResourceId, noEcho) {
+  return new Promise((resolve, reject) => {
+    var responseBody = JSON.stringify({
+      Status: responseStatus,
+      Reason: \"See the details in CloudWatch Log Stream: \" + context.logGroupName + \" \" + context.logStreamName,
+      PhysicalResourceId: physicalResourceId || context.logStreamName,
+      StackId: event.StackId,
+      RequestId: event.RequestId,
+      LogicalResourceId: event.LogicalResourceId,
+      NoEcho: noEcho || false,
+      Data: responseData
+    });
+    
+    console.log(\"Response body:\", responseBody);
+    
+    var https = require(\"https\");
+    var url = require(\"url\");
+    
+    var parsedUrl = url.parse(event.ResponseURL);
+    var options = {
+      hostname: parsedUrl.hostname,
+      port: 443,
+      path: parsedUrl.path,
+      method: \"PUT\",
+      headers: {
+        \"content-type\": \"\",
+        \"content-length\": responseBody.length
+      }
+    };
+    
+    var request = https.request(options, function(response) {
+      console.log(\"Status code: \" + response.statusCode);
+      console.log(\"Status message: \" + response.statusMessage);
+      resolve();
+    });
+    
+    request.on(\"error\", function(error) {
+      console.log(\"respond(..) failed executing https.request(..): \" + error);
+      resolve();
+    });
+    
+    request.write(responseBody);
+    request.end();
+  });
+};
+
+const AWS = require('aws-sdk');
+
+exports.handler = async function (event, context) {
+  console.log(JSON.stringify(event, null, 4));
+  try {
+    const projectName = event.ResourceProperties.ProjectName;
+    const codebuild = new AWS.CodeBuild();
+    
+    console.log(\`Starting new build of project \${projectName}\`);
+    
+    const { build } = await codebuild.startBuild({
+      projectName,
+      // Pass CFN related parameters through the build for extraction by the
+      // completion handler.
+      buildspecOverride: event.RequestType === 'Delete' ? 'buildspec-destroy.yml' : 'buildspec-deploy.yml',
```

Comment:
The important change is here  : no more override of buildspec but use of the one in the repo ... 
Advantage being, the consumer can easily change it's install/build/deploy process independently of the oneClickDeployer ... 

Disadvantage: the consumer needs to have those files in in CDK app repo ...